uaclient/security_status.py

```diff
@@ -191,3 +202,276 @@ def security_status(cfg: UAConfig) -> Dict[str, Any]:
+                pkg_txt=package_text,
+            )
+        )
+
+
+def _print_info_messages(summary: Dict[str, Any]) -> None:
+    if summary["num_third_party_packages"]:
+        print("")
+        print(messages.SS_THIRD_PARTY)
+        print(messages.SS_MORE_INFO.format(option="--thirdparty"))
+
+    if summary["num_unknown_packages"]:
+        print("")
+        print(messages.SS_UNAVAILABLE)
+        print(messages.SS_MORE_INFO.format(option="--unavailable"))
+
+
+def _print_esm_actions(
+    summary: Dict[str, Any],
+    esm_infra_status: ApplicationStatus,
+    esm_apps_status: ApplicationStatus,
+) -> None:
+    updates_esm_infra = summary["num_esm_infra_updates"]
+    updates_esm_apps = summary["num_esm_apps_updates"]
+    packages_infra = (
+        summary["num_main_packages"]
+        + summary["num_restricted_packages"]
+        + summary["num_esm_infra_packages"]
+    )
+    packages_apps = (
+        summary["num_universe_packages"]
+        + summary["num_multiverse_packages"]
+        + summary["num_esm_apps_packages"]
+    )
+
+    is_attached = summary["ua"]["attached"]
+    is_lts = is_current_series_lts()
+
+    if is_lts and esm_infra_status == ApplicationStatus.DISABLED:
+        if not is_supported(series):
+            print("")
+            print(
+                messages.SS_SERVICE_ADVERTISE.format(
+                    service="Ubuntu Pro with ESM Infra",
+                    updates=updates_esm_infra,
+                    pl_updates="s" if updates_esm_infra > 1 else "",
+                    packages=packages_infra,
+                    pl_packages="s" if packages_infra > 1 else "",
+                )
+            )
+            if is_attached:
+                print("")
+                print(
+                    messages.SS_SERVICE_COMMAND.format(
+                        service="ESM Infra", command="ua enable esm-infra"
+                    )
+                )
+
+    if (
+        is_lts
+        and packages_apps
+        and esm_apps_status == ApplicationStatus.DISABLED
+    ):
+        print("")
+        print(
+            messages.SS_SERVICE_ADVERTISE.format(
+                service="Ubuntu Pro with ESM Apps",
+                updates=updates_esm_apps,
+                pl_updates="s" if updates_esm_apps > 1 else "",
+                packages=packages_apps,
+                pl_packages="s" if packages_apps > 1 else "",
+            )
+        )
+        if is_attached:
+            print("")
+            print(
+                messages.SS_SERVICE_COMMAND.format(
+                    service="ESM Apps", command="ua enable esm-apps"
+                )
+            )
+
+    if is_lts and not is_attached:
+        print("")
+        print(messages.SS_UNATTACHED)
+
+
+def security_status(cfg: UAConfig):
+    packages_summary = security_status_dict(cfg)["summary"]
+
+    esm_infra_status = ESMInfraEntitlement().application_status()[0]
+    esm_apps_status = ESMAppsEntitlement().application_status()[0]
```

Comment:
Let's pass the `cfg` object when creating new entitlement objects, otherwise we will create another `UAConfig` instance each time